File: java/src/test/java/javasearch/FileUtilTest.java
package javasearch;

import org.junit.Test;

import java.io.File;
import java.util.List;

import static org.junit.Assert.*;

public class FileUtilTest {
    public FileUtilTest() {}

    /***************************************************************************
     * getExtension tests
     **************************************************************************/
    @Test
    public void testGetTxtExtension() {
        File file = new File("filename.txt");
        assertEquals(FileUtil.getExtension(file), "txt");
    }

    @Test
    public void testGetMissingExtension() {
        File file = new File("filename.");
        assertEquals(FileUtil.getExtension(file), "");
    }

    @Test
    public void testGetNoExtension() {
        File file = new File("filename");
        assertEquals(FileUtil.getExtension(file), "");
    }

    @Test
    public void testGetHiddenTxtExtension() {
        File file = new File(".filename.txt");
        assertEquals(FileUtil.getExtension(file), "txt");
    }

    @Test
    public void testGetHiddenMissingExtension() {
        File file = new File("filename.");
        assertEquals(FileUtil.getExtension(file), "");
    }

    @Test
    public void testGetHiddenNoExtension() {
        File file = new File("filename");
        assertEquals(FileUtil.getExtension(file), "");
    }

    /***************************************************************************
     * isDotDir tests
     **************************************************************************/
    @Test
    public void testIsDotDirSingleDot() {
        String filename = ".";
        assert(FileUtil.isDotDir(filename));
    }

    @Test
    public void testIsDotDirDoubleDot() {
        String filename = "..";
        assert(FileUtil.isDotDir(filename));
    }

    @Test
    public void testIsDotDirNotDotDir() {
        String filename = "~/path";
        assert(!FileUtil.isDotDir(filename));
    }

    @Test
    public void testIsDotDirPathWithDot() {
        String filename = "./path";
        assert(!FileUtil.isDotDir(filename));
    }

    @Test
    public void testIsDotDirHiddenFile() {
        String filename = ".gitignore";
        assert(!FileUtil.isDotDir(filename));
    }

    /***************************************************************************
     * isHidden tests
     **************************************************************************/
    @Test
    public void testIsHiddenSingleDot() {
        String filename = ".";
        assert(!FileUtil.isHidden(filename));
    }

    @Test
    public void testIsHiddenDoubleDot() {
        String filename = "..";
        assert(!FileUtil.isHidden(filename));
    }

    @Test
    public void testIsHiddenHiddenFileName() {
        String filename = ".gitignore";
        assert(FileUtil.isHidden(filename));
    }

    @Test
    public void testIsHiddenNotHiddenFileName() {
        String filename = "file.txt";
        assert(!FileUtil.isHidden(filename));
    }

    @Test
    public void testIsHiddenHiddenFile() {
        File file = new File(".gitignore");
        assert(FileUtil.isHidden(file));
    }

    @Test
    public void testIsHiddenNotHiddenFile() {
        File file = new File("./file.txt");
        assert(!FileUtil.isHidden(file));
    }

    /***************************************************************************
     * splitPath tests
     **************************************************************************/
    @Test
    public void testSplitPathWithDot() {
        String path = "./path/to/somewhere/";
        List<String> elems = FileUtil.splitPath(path);
        assert(elems.size() == 3);
        assert(elems.get(0).equals("path"));
    }

    @Test
    public void testSplitPathWithDoubleDot() {
        String path = "../path/to/somewhere/";
        List<String> elems = FileUtil.splitPath(path);
        assert(elems.size() == 3);
        assert(elems.get(0).equals("path"));
    }

    @Test
    public void testSplitPathWithoutDot() {
        String path = "/path/to/somewhere/";
        List<String> elems = FileUtil.splitPath(path);
        assert(elems.size() == 3);
        assert(elems.get(0).equals("path"));
    }
}

File: typescript/tssearch/tests/fileutil_test.ts
/*
 * fileutil_test.js
 *
 * Some nodeunit tests of fileutil.js
 */

import {FileUtil} from '../src/fileutil';

/***************************************************************************
 * getExtension tests
 **************************************************************************/
exports.testGetTxtExtension = function(test) {
    var file: string = "filename.txt";
    test.ok(FileUtil.getExtension(file) === "txt", "ext == \"txt\"");
    test.done();
};

exports.testGetMissingExtension = function(test) {
    var file: string = "filename.";
    test.ok(FileUtil.getExtension(file) === "", "ext == \"\"");
    test.done();
};

exports.testGetNoExtension = function(test) {
    var file: string = "filename";
    test.ok(FileUtil.getExtension(file) === "", "ext == \"\"");
    test.done();
};

exports.testGetHiddenTxtExtension = function(test) {
    var file: string = ".filename.txt";
    test.ok(FileUtil.getExtension(file) === "txt", "ext == \"txt\"");
    test.done();
};

exports.testGetHiddenMissingExtension = function(test) {
    var file: string = "filename.";
    test.ok(FileUtil.getExtension(file) === "", "ext == \"\"");
    test.done();
};

exports.testGetHiddenNoExtension = function(test) {
    var file: string = "filename";
    test.ok(FileUtil.getExtension(file) === "", "ext == \"\"");
    test.done();
};

/***************************************************************************
 * getRelativePath tests
 **************************************************************************/
exports.testGetRelativePath = function(test) {
    var filepath: string = "/Users/cary/filename.txt";
    test.ok(FileUtil.getRelativePath(filepath, '.') === "./filename.txt", "relativepath === ./filenametxt");
    test.done();
};

/***************************************************************************
 * isDotDir tests
 **************************************************************************/
exports.testIsDotDirSingleDot = function(test) {
    var filename: string = ".";
    test.ok(FileUtil.isDotDir(filename), "isDotDir(" + filename + ") == true");
    test.done();
};

exports.testIsDotDirDoubleDot = function(test) {
    var filename: string = "..";
    test.ok(FileUtil.isDotDir(filename), "isDotDir(" + filename + ") == true");
    test.done();
};

exports.testIsDotDirNotDotDir = function(test) {
    var filename: string = "~/path";
    test.ok(!FileUtil.isDotDir(filename), "isDotDir(" + filename + ") == false");
    test.done();
};

exports.testIsDotDirPathWithDot = function(test) {
    var filename: string = "./path";
    test.ok(!FileUtil.isDotDir(filename), "isDotDir(" + filename + ") == false");
    test.done();
};

exports.testIsDotDirHiddenFile = function(test) {
    var filename: string = ".gitignore";
    test.ok(!FileUtil.isDotDir(filename), "isDotDir(" + filename + ") == false");
    test.done();
};

/***************************************************************************
 * isHidden tests
 **************************************************************************/
exports.testIsHiddenSingleDot = function(test) {
    const filename: string = ".";
    test.ok(!FileUtil.isHidden(filename));
    test.done();
};

exports.testIsHiddenDoubleDot = function(test) {
    const filename: string = "..";
    test.ok(!FileUtil.isHidden(filename));
    test.done();
};

exports.testIsHiddenHiddenFile = function(test) {
    const filename: string = ".gitignore";
    test.ok(FileUtil.isHidden(filename));
    test.done();
};

exports.testIsHiddenNotHiddenFile = function(test) {
    const filename: string = "file.txt";
    test.ok(!FileUtil.isHidden(filename));
    test.done();
};

/***************************************************************************
 * expandPath tests
 **************************************************************************/
exports.testExpandPathPathWithTilde = function(test) {
    const filepath: string = "~/filename.txt";
    const expected: string = process.env.HOME + "/filename.txt";
    test.ok(FileUtil.expandPath(filepath) === expected);
    test.done();
};

exports.testExpandPathPathNoTilde = function(test) {
    const filepath: string = "./filename.txt";
    test.ok(FileUtil.expandPath(filepath) === filepath);
    test.done();
};
